### PR TITLE
Categorizing internal messages, limiting to one internal message type per session and removing properties from internal messages

### PR DESCRIPTION
**Pull request by AlexBulankou**
- Creating a distinct type for each internal message
- Limiting each message type to one per session
- Not setting any properties on internal messages

### CODE REVIEW

**Comment by AlexBulankou** [comment]
This will address #125

**Comment by madeleinefletcher** [comment]
:shipit:

File: JavaScript/JavaScriptSDK.Tests/CheckinTests/Logging.tests.ts [hunk 1]
```diff
@@ -73,17 +76,17 @@ class LoggingTests extends TestClass {
                 InternalLogging.verboseLogging = () => true;
 
                 // act
-                InternalLogging.throwInternalNonUserActionable(Microsoft.ApplicationInsights.LoggingSeverity.WARNING, new InternalLoggingMessage("error!"));
-                InternalLogging.throwInternalUserActionable(Microsoft.ApplicationInsights.LoggingSeverity.WARNING, new InternalLoggingMessage("error!"));
-                InternalLogging.throwInternalNonUserActionable(Microsoft.ApplicationInsights.LoggingSeverity.CRITICAL, new InternalLoggingMessage("error!"));
-                InternalLogging.throwInternalUserActionable(Microsoft.ApplicationInsights.LoggingSeverity.CRITICAL, new InternalLoggingMessage("error!"));
+                InternalLogging.throwInternalNonUserActionable(Microsoft.ApplicationInsights.LoggingSeverity.WARNING, new InternalLoggingMessage(3, "error!"));
```
Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/Logging.tests.ts]: Numbers "3", "4", etc look confusing. Does that matter for the test? if yes, can you do global 
``var i = 0;``
 and then pass 
``++i``
as the 1st arg in all tests?

**Comment by MaxShehovtsov** [comment]
Nicely done! :shipit:

File: JavaScript/JavaScriptSDK.Tests/CheckinTests/Logging.tests.ts (hunk 1 above)
Comment (AlexBulankou) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/Logging.tests.ts]: Yes, the reason I'm using different _InternalMessageIDs it is get them all to be sent. I'll change to ++i.

Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/Logging.tests.ts]: So there's a slight chance you hit internal throttling for tests :) You should reset it between tests somehow

Comment (AlexBulankou) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/Logging.tests.ts]: Yes, line 19 in Logging.tests.ts

**Pull request merged**